spec/models/pawn_spec.rb

```diff
@@ -0,0 +1,203 @@
+require 'rails_helper'
+
+RSpec.describe Pawn, type: :model do
+  describe "#valid_move?" do
+    context "white moving" do
+      it "should be true when moving 1 space forward" do
+        board = create(:game)
+        board.pieces.delete_all
+        pawn = Pawn.create(x_position: 1,
+                           y_position: 1,
+                           game_id: board.id,
+                           color: true)
+        board.reload
+        actual = pawn.valid_move?(1, 2)
+        expect(actual).to be(true)
+      end
+      it "should be true when moving 2 spaces forward on first move" do
+        board = create(:game)
+        board.pieces.delete_all
+        pawn = Pawn.create(x_position: 1,
+                           y_position: 1,
+                           game_id: board.id,
+                           color: true)
+        board.reload
+        actual = pawn.valid_move?(1, 3)
+        expect(actual).to be(true)
+      end
+      it "should be false when moving 3 space forward" do
+        board = create(:game)
+        board.pieces.delete_all
+        pawn = Pawn.create(x_position: 1,
+                           y_position: 1,
+                           game_id: board.id,
+                           color: true)
+        board.reload
+        actual = pawn.valid_move?(1, 4)
+        expect(actual).to be(false)
+      end
+      it "should be false when moving 2 spaces forward after first move" do
+        board = create(:game)
+        board.pieces.delete_all
+        pawn = Pawn.create(x_position: 1,
+                           y_position: 2,
+                           game_id: board.id,
+                           color: true)
+        board.reload
+        actual = pawn.valid_move?(1, 4)
+        expect(actual).to be(false)
+      end
+      it "should be false when moving 1 space backward" do
+        board = create(:game)
+        board.pieces.delete_all
+        pawn = Pawn.create(x_position: 1,
+                           y_position: 4,
+                           game_id: board.id,
+                           color: true)
+        board.reload
+        actual = pawn.valid_move?(1, 3)
+        expect(actual).to be(false)
+      end
+      it "should be true when moving 1 space diagonal/left when capturing" do
```

Comment:
so this test passes fine, but should we test when a pawn tries to move diagonally to a same-color piece? currently this will cause a runtime error in the move_to! method (so it would eventually get caught), but we may just want to catch that in valid_move? to prevent it from getting that far. see @amarkpark's knight valid_move? implementation for an example....we may also want to update all the other pieces that currently don't check this